DataSourceVMwareGuestInfo.py

```diff
@@ -268,6 +273,61 @@ def get_guestinfo_value(key):
     return None
 
 
+def set_guestinfo_value(key, value):
+    '''
+    Sets a guestinfo value for the specified key. Set value to an empty string
+    to clear an existing guestinfo key.
+    '''
+
+    # If value is an empty string then set it to a single space as it is not
+    # possible to set a guestinfo key to an empty string. Setting a guestinfo
+    # key to a single space is as close as it gets to clearing an existing
+    # guestinfo key.
+    if value == "":
+        value = " "
```

Comment:
A note for this change. Setting it to a single space doesn't break cloud-init when cloud-init runs afterwards, because cloud-init-vmware-guestinfo strips the spaces :

- https://github.com/vmware/cloud-init-vmware-guestinfo/blob/b832d4a0cc8ec359249a233dbb61f888116b7de0/DataSourceVMwareGuestInfo.py#L257
- https://github.com/vmware/cloud-init-vmware-guestinfo/blob/b832d4a0cc8ec359249a233dbb61f888116b7de0/DataSourceVMwareGuestInfo.py#L277

Reply:
Hi @jessehu,

I'm curious by what you mean by:

> Setting it to a single space doesn't break cloud-init 

Did you mean it **does** break the datasource? Or if you meant what you said, is there a reason you think it's worth calling out the intended behavior?

FWIW, [the data isn't cleaned up until **after** it's read](https://github.com/akutz/cloud-init-vmware-guestinfo/blob/ee087279dc3231b8c1e4372d392aade0769594a3/DataSourceVMwareGuestInfo.py#L114-L125):

```python
        # Get the metadata.
        self.metadata = load_metadata()

        # Get the user data.
        self.userdata_raw = guestinfo('userdata')

        # Get the vendor data.
        self.vendordata_raw = guestinfo('vendordata')

        # Check to see if any of the guestinfo data should be removed.
        if data_access_method == VMWARE_RPCTOOL:
            clear_guestinfo_keys(self.metadata['cleanup-guestinfo'])
```

So while I'm not sure what your intent was with your original comment, I think the PR's design is okay based on the data workflow model. Do you agree, or am I missing something super obvious?